src/Servers/IIS/src/Microsoft.AspNetCore.Server.IIS/Core/IISHttpServer.cs

```diff
@@ -84,6 +83,9 @@ public Task StartAsync<TContext>(IHttpApplication<TContext> application, Cancell
 
             _iisContextFactory = new IISContextFactory<TContext>(_memoryPool, application, _options, this, _logger);
             _nativeApplication.RegisterCallbacks(_requestHandler, _shutdownHandler, _onDisconnect, _onAsyncCompletion, (IntPtr)_httpServerHandle, (IntPtr)_httpServerHandle);
+
+            Features.Set<IServerAddressesFeature>(new ServerAddressesFeature(_options.ServerAddresses));
```

Comment:
Create the feature in ctor, then populate it here in start. Include a comment that this is for parity with kestrel & httpsys.